### ISSUE TITLE
Push not working in tab based app with many tabs [iOS]

### ISSUE THREAD

**Issue by fdwcomune** [wix/react-native-navigation#1865]
### Issue Description

In a tab based app (**startTabBasedApp**) with more than 5(?) tabs **navigator.push** does not show a new screen when called inside the 5th or higher tabs. Works correctly in tabs 1 - 4.

##### 1) Create a tab based app with more than 5 tabs:

```
Navigation.startTabBasedApp({
	tabs: [
		{
			icon: ICON_1,
			label: 'Tab 1',
			screen: 'app.Screen1',
			title: 'Tab 1',
		},
		{
			icon: ICON_2,
			label: 'Tab 2',
			screen: 'app.Screen2',
			title: 'Tab 2',
		},
		{
			icon: ICON_3,
			label: 'Tab 3',
			screen: 'app.Screen3',
			title: 'Tab 3',
		},
		{
			icon: ICON_4,
			label: 'Tab 4',
			screen: 'app.Screen4',
			title: 'Tab 4',
		},
		{
			icon: ICON_5,
			label: 'Tab 5',
			screen: 'app.Screen5',
			title: 'Tab 5',
		},
		{
			icon: ICON_6,
			label: 'Tab 6',
			screen: 'app.Screen6',
			title: 'Tab 6',
		},
	],
});
```

##### 2) Push a screen
Within **Screen5** or **Screen6**

```
this.props.navigator.push({
	screen: 'app.AnotherScreen',
	title: 'Another Screen',
});
```

#### Outcome

Nothing seems to happen, **app.AnotherScreen** is not shown/visible, however debugger console outputs the following

```
RCTLog.js:43 Running application app.AnotherScreen ({
    initialProps =     {
        commandType = Push;
        navigatorEventID = "screenInstanceID12_events";
        navigatorID = "controllerID3_nav5";
        screenInstanceID = screenInstanceID12;
        timestamp = 1505461683043;
    };
    rootTag = 91;
})
infoLog.js:17 Running application "app.AnotherScreen" with appParams: {"rootTag":91,"initialProps":{"screenInstanceID":"screenInstanceID12","timestamp":1505461683043,"navigatorID":"controllerID3_nav5","navigatorEventID":"screenInstanceID12_events","commandType":"Push"}}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF
```
---
### Environment
* React Native Navigation version: 1.1.220
* React Native version: 0.47.1
* Platforms: iOS
* Device info: Simulator (iOS 10.3) as well as various devices (iOS 10.3)

**Comment by dbelchev** [comment]
Same for android.

**Comment by Shelf-A** [comment]
@fdwcomune did you manage to solve this?

**Comment by fdwcomune** [comment]
I worked around the issue by implementing a custom "tab based app". I haven't checked the latest version of react-native-navigation though, it might be fixed(?)

I made each tab of my app a [single screen app](https://wix.github.io/react-native-navigation/#/top-level-api?id=startsinglescreenappparams) and made a custom bottom bar that started the current single screen tab when "switching tab". I added the the custom bottom bar to all my screens.

This worked OK but the screen does flash white every time the user switches tabs.
